app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -59,6 +60,24 @@ open class TwoPaneFragment : Fragment() {
 
     private fun observeThreadEvents() = with(mainViewModel) {
 
+        val threadListAdapter = when (this@TwoPaneFragment) {
+            is ThreadListFragment -> this@TwoPaneFragment.threadListAdapter
+            is SearchFragment -> this@TwoPaneFragment.threadListAdapter
+            else -> null
+        }
+
+        // Reset selected Thread UI when closing Thread
+        currentThreadUid.observe(viewLifecycleOwner) { threadUid ->
```

Comment:
If you don't use the value `threadUid`, you can just return a `boolean` to know if this value is `null`.

Reply:
The `threadUid` value is used in another `observe`.